classes/file/TemporaryFileManager.inc.php

```diff
@@ -57,6 +57,16 @@ function deleteFile($fileId, $userId) {
 	}
 
 	/**
+	 * Delete a file by path.
+	 * For temporary files that are not in the DB, e.g. exported xml files.
+	 * @param $filePath string the location of the file to be deleted
+	 * @return boolean returns true if successful
+	 */
+	function deleteFileByPath($filePath) {
```

Comment:
Why not let the TemporaryFileManager persist the file in the database? Then you could get rid of `deleteFileByPath`, `downloadFileByPath`, and even `ImportExportPlugin::getExportFileName`, and let the TemporaryFileManager handle more of the details.